docs/core/sdk/file-based-apps.md

```diff
@@ -0,0 +1,406 @@
+
+Key benefits include:
+
+- Reduced boilerplate for simple applications.
+- Self-contained source files with embedded configuration.
+- Native AOT publishing enabled by default.
+- Automatic packaging as .NET tools.
+
+In this article, learn how to create, configure, and work with file-based apps effectively.
+
+## Supported directives
+
+File-based apps use directives prefixed with `#:` to configure the build and run the application. Supported directives include: `#:package`, `#:project`, `#:property`, and `#:sdk`. Place these directives at the top of the C# file.
+
+### `#:package`
+
+Adds a NuGet package reference to your application.
+
+```csharp
+#:package Newtonsoft.Json
+#:package Serilog version="3.1.1"
+```
+
+### `#:project`
+
+References another project file or directory that contains a project file.
+
+```csharp
+#:project ../SharedLibrary/SharedLibrary.csproj
+```
+
+### `#:property`
+
+Sets an MSBuild property value.
+
+```csharp
+#:property TargetFramework=net10.0
+#:property PublishAot=false
+```
+
+### `#:sdk`
+
+Specifies the SDK to use. Defaults to `Microsoft.NET.Sdk`.
+
+```csharp
+#:sdk Microsoft.NET.Sdk.Web
+```
+
+## CLI commands
+
+The .NET CLI provides full support for file-based apps through familiar commands.
+
+### Run applications
+
+Run a file-based app directly by using the `dotnet run` command:
+
+```dotnetcli
+dotnet run file.cs
+```
+
+Or use the shorthand syntax:
+
+```dotnetcli
+dotnet file.cs
+```
+
+#### Pass arguments
+
+Pass arguments to your application in several ways:
+
+```dotnetcli
+dotnet run file.cs -- arg1 arg2
+```
+
+Arguments after `--` are passed to your application. Without `--`, arguments go to the `dotnet run` command:
+
+```dotnetcli
+dotnet run file.cs arg1 arg2
+```
+
+However, with the shorthand syntax, all arguments go to your application:
+
+```dotnetcli
+dotnet file.cs arg1 arg2
+```
+
+### Build applications
+
+Compile your file-based app by using the `dotnet build` command:
+
+```dotnetcli
+dotnet build file.cs
+```
+
+The SDK generates a temporary project and builds your application.
+
+### Clean build outputs
+
+Remove build artifacts by using the `dotnet clean` command:
+
+```dotnetcli
+dotnet clean file.cs
+```
+
+Clean all file-based apps in a directory:
+
+```dotnetcli
+dotnet clean file-based-apps
+```
+
+### Publish applications
+
+Create a deployment package by using the `dotnet publish` command:
+
+```dotnetcli
+dotnet publish file.cs
+```
+
+File-based apps enable native AOT publishing by default, producing optimized, self-contained executables.
+
+### Package as tool
+
+Package your file-based app as a .NET tool by using the `dotnet pack` command:
+
+```dotnetcli
+dotnet pack file.cs
+```
+
+File-based apps set `PackAsTool=true` by default.
```

Comment:
Consider following this with a line showing how to disable this, e.g.:


```suggestion
File-based apps set `PackAsTool=true` by default. This can be disabled by placing `#:property PackAsTool=false` at the top of your file.
```